node/network/availability-recovery/src/lib.rs

```diff
@@ -569,8 +569,20 @@ const fn is_unavailable(
 }
 
 /// Re-encode the data into erasure chunks in order to verify
-/// the root hash of the provided merkle tree, which is built
+/// the root hash of the provided Merkle tree, which is built
 /// on-top of the encoded chunks.
+///
+/// This (expensive) check is necessary, as otherwise we can't be sure that some chunk's won't have
+/// been tampered with by the backers, which would result in some validators voting valid and some
```

Comment:
I would avoid discussion of 'voting' here, it's out of context. I would describe it as 'recovering successfully or not'